internal/ctlog/ctlog.go

```diff
@@ -834,6 +862,83 @@ func (l *Log) sequencePool(ctx context.Context, p *pool) (err error) {
 
 var testingOnlyPauseSequencing func()
 
+type uploadAction struct {
+	key  string
+	data []byte
+	opts *UploadOptions
+}
+
+func marshalStagedUploads(uploads []*uploadAction) ([]byte, error) {
+	var buffer bytes.Buffer
+	writer := tar.NewWriter(&buffer)
```

Comment:
Since we are both encoder and decoder, and no external system is expected to consume these, tar is fine.

If we need to version, we can change the path.